wandb/sdk/launch/agent/agent.py

```diff
@@ -595,7 +606,18 @@ def _thread_run_job(
             return
         with self._jobs_lock:
             job_tracker.run = run
+        start_time = time.time()
         while self._jobs_event.is_set():
+            # If run has failed to start before timeout, kill it
+            state = run.get_status().state
+            if state == "starting" and RUN_START_TIMEOUT > 0:
+                if time.time() - start_time > RUN_START_TIMEOUT:
+                    run.cancel()
+                    raise LaunchError(
```

Comment:
Did we test this path with k8s? Would be great to make sure we see this working before we ship

Reply:
Alternatively, could add a system test for this

Reply:
Yeah I have tested this with k8s but I agree a system test would be great